pkg/adapter/containers.go

```diff
@@ -1058,7 +1058,14 @@ func (r *LocalRuntime) GenerateSystemd(c *cliconfig.GenerateSystemdValues) (stri
 	if c.Name {
 		name = ctr.Name()
 	}
-	return systemdgen.CreateSystemdUnitAsString(name, ctr.ID(), c.RestartPolicy, ctr.Config().StaticDir, timeout)
+
+	config := ctr.Config()
+	conmonPidFile := config.ConmonPidFile
+	if conmonPidFile == "" {
+		return "", errors.Errorf("conmon PID file path is empty, try to recreate the container with --conmon-pidfile flag")
```

Comment:
there is a check in `runtime_ctr.go` (inside `setupContainer`) for assigning a value to ConmonPidFile when it is empty.  We do that only for rootless currently.  Could change that so it is done also for root containers?

Reply:
> We can expose the path with a function and use it in `generate systemd`

This _seems_ to work in practice, but I not sure that completely correctly, as `systemd` issues a warning if you try to make it supervise not an immediate child (guess this is because in this case `systemd` will not to receive `SIGCHLD` for specified PID on its death). For non-child processes, `systemd` polls the process state from `/proc` as a workaround, and this is probably not so reliable (e.g. a different process with same PID may kick in between polls).

I think that waiting for `conmon` is a more correct approach here, and we probably rather should make sure that `conmon` always has a pidfile (e.g. set a default location), as @giuseppe proposed.

Reply:
Done.

I left the error on its place for the case when user would try to generate a unit file for container from old Podman without ConmonPidFile set.